sylvia-derive/src/querier.rs

```diff
@@ -34,133 +36,54 @@ where
         let Self {
             variants,
             associated_types,
-            ..
+            interface_name,
         } = self;
 
         let generics: Vec<_> = associated_types
             .without_special()
             .map(ItemType::as_name)
             .collect();
-        let methods_impl = variants
+
+        let accoc_types: Vec<_> = associated_types
+            .without_special()
+            .map(ItemType::as_name)
+            .map(|assoc| quote! {Self:: #assoc})
+            .collect();
+        let methods_trait_impl = variants
             .variants()
             .iter()
-            .map(|variant| variant.emit_trait_querier_impl(&generics));
+            .map(|variant| variant.emit_trait_querier_impl(&accoc_types));
+
+        let methods_trait_impl_copy = methods_trait_impl.clone();
 
         let methods_declaration = variants
             .variants()
             .iter()
             .map(|variant| variant.emit_querier_declaration(&generics));
 
-        let types_declaration = associated_types.filtered();
-        let types_definition = associated_types.emit_types_definition();
+        let types_declaration = associated_types.filtered().cloned();
         let where_clause = associated_types.as_where_clause();
 
         quote! {
-            pub struct BoundQuerier<'a, C: #sylvia ::cw_std::CustomQuery, #(#generics,)* > {
-                contract: &'a #sylvia ::cw_std::Addr,
-                querier: &'a #sylvia ::cw_std::QuerierWrapper<'a, C>,
-                _phantom: std::marker::PhantomData<( #(#generics,)* )>,
-            }
-
-            impl<'a, C: #sylvia ::cw_std::CustomQuery, #(#generics,)* > BoundQuerier<'a, C, #(#generics,)* > {
-                pub fn querier(&self) -> &'a #sylvia ::cw_std::QuerierWrapper<'a, C> {
-                    self.querier
-                }
-
-                pub fn contract(&self) -> &'a #sylvia ::cw_std::Addr {
-                    self.contract
-                }
-
-                pub fn borrowed(contract: &'a #sylvia ::cw_std::Addr, querier: &'a #sylvia ::cw_std::QuerierWrapper<'a, C>) -> Self {
-                    Self { contract, querier, _phantom: std::marker::PhantomData }
-                }
-            }
-
-            impl <'a, C: #sylvia ::cw_std::CustomQuery, #(#generics,)* > Querier for BoundQuerier<'a, C, #(#generics,)* > #where_clause {
-                #(#types_definition)*
-
-                #(#methods_impl)*
-            }
+            pub use #sylvia ::types::{BoundQuerier, Remote};
+            use std::marker::PhantomData;
```

Comment:
Don't use imports in generated code. Instead use full path where needed.

Reply:
Done